### PR TITLE
DATA-600 Support _REMOVE_FILTER_ for all filter types

### PR DESCRIPTION
**Pull request by ShilpaSivanesan**

### CODE REVIEW

**Comment by ShilpaSivanesan** [comment]
@Gauravshah  Moving the filter condition outside so that it supports both IN and JSON Filters
We need multiple return statements as to figure out filter type

File: dist/datasource.js [hunk 1]
```diff
@@ -365,6 +356,18 @@ function (angular, _, dateMath, moment) {
...
       if (replacedFilters) {
+        replacedFilters = replacedFilters.filter(function (filter) {
+          if (filter.pattern)
+            return filter.pattern != "_REMOVE_FILTER_"
+          else if (filter.value)
+            return filter.value != "_REMOVE_FILTER_"
+          else if (filter.values)
+            return !filter.values.includes("_REMOVE_FILTER_")
+          else
+            return true
```
Comment (Gauravshah) [inline dist/datasource.js]: we are missing the nested section

File: dist/datasource.js [hunk 2]
```diff
@@ -98,536 +98,573 @@ function (angular, _, dateMath, moment) {
...
+                  finalFilter = JSON.parse(filter.value);
+                }
+                return finalFilter;
+              });
+
+          if (replacedFilters) {
+
+            replacedFilters = replacedFilters.filter(function (filter) {
+
+              if(filter.fields){
```
Comment (Gauravshah) [inline dist/datasource.js]: this condition code is being repeated. we can avoid as follows
```suggestion
if (replacedFilters) {
   replacedFilters = remove_filter_recursively(replacedFilters)
}
function remove_filter_recursively(filters){
    return filters.filter(function (filter) {
        if(filter.fields) {
            remove_filter_recursively(filter.fields)
        }
        else if (filter.pattern)
            return filter.pattern !== "_REMOVE_FILTER_"
        else if (filter.value)
            return filter.value !== "_REMOVE_FILTER_"
        else if (filter.values)
            return !filter.values.includes("_REMOVE_FILTER_")
        else
            return true
    })     
}

```

**Comment by Gauravshah** [comment]
opened another  PR : https://github.com/GoshPosh/druidplugin/pull/23

**Pull request closed without merging**